### ISSUE TITLE
[Discovery] Add pagination support

### ISSUE THREAD

**Issue by duglin** [cloudevents/spec#657]
The list of services returned from our queries might be large - we should add the notion of pagination to our queries.

**Comment by duglin** [comment]
Things to consider:
- client specifying the # of items per pull?
- server indicating where to get the next chunk from - part of data vs HTTP headers?
- does the server need to maintain a static "result set" for the lifetime of the set of pulls per client?

e.g. As an HTTP header:
```
Link: <https://api.github.com/user/repos?page=3&per_page=100>; rel="next",
  <https://api.github.com/user/repos?page=50&per_page=100>; rel="last"
```

- "page" vs "next"
```
  - example.com?page=2&size=100    # does start of page 2 change based on size?
  - example.com?next=[opaque string]&size=100   #  "next" doesn't change based on size
     just changes the current page size (but will influence what the returned "next" value will be)
```

**Comment by duglin** [comment]
Closing since we merged #668
